tests/cpp/test_matmul.cpp

```diff
@@ -3775,12 +3800,28 @@ TEST_F(HopperMatmulTest, HSH_NT_128BSwizzle) {
     tv2->axis(-3)->parallelize(ParallelType::Mma);
   }
 
-  for (auto tv : {tv3c, tv3}) {
+  if (!use_smem_epilogue) {
+    for (auto tv : {tv3c, tv3}) {
+      auto s = mma_utils::MmaSwizzler::scheduleMmaOutputAllocation(
+          tv->getLoopDomain());
+      tv->setLoopDomain(s.as<IterDomain*>());
+    }
+  } else {
     auto s = mma_utils::MmaSwizzler::scheduleMmaOutputAllocation(
-        tv->getLoopDomain());
-    tv->setLoopDomain(s.as<IterDomain*>());
+        tv3c->getLoopDomain());
+    tv3c->setLoopDomain(s.as<IterDomain*>());
+    tv3c->setAllocationDomain(s.as<IterDomain*>(), true);
```

Comment:
Is this necessary? The loop domain was just set so shouldnt that be the default allocation domain?

Reply:
This is require - I don't understand this well, maybe @zasdfgbnm can tell us the exact reason.

Reply:
In the legacy indexer, the default allocation is not the loop domain, it is the `getMaybeAllocationDomain`, which is the logical domain. So the behavior isn't really well-defined in the transition of indexers... My suggestion is, if commenting this out this test still pass, then remove this. Otherwise, keep it.

Reply:
Without it, the test doesn't pass